src/main/java/com/diffmin/patch/PatchApplication.java

```diff
@@ -77,6 +77,28 @@ private static void performInsertion(
                                 ((CtExecutable<?>) toBeInserted.getParent()).getThrownTypes());
                 ((CtExecutable<?>) inWhichElement).setThrownTypes(thrownTypesCopy);
                 break;
+            case CONTAINED_TYPE:
+                // Inserting into CtPackage
+                Set<CtType<?>> typesCopy =
+                        new HashSet<>(((CtPackage) toBeInserted.getParent()).getTypes());
+                for (CtType<?> type : typesCopy) {
+                    type.setParent(inWhichElement);
+                }
```

Comment:
```suggestion
                Set<CtType<?>> typesCopy = ((CtType<?>) toBeInserted).getPackage().getTypes().stream()
                        .map(CtType::clone)
                        .collect(Collectors.toSet());
                ((CtPackage) inWhichElement).setTypes(typesCopy);
```

Reply:
I wish, one day, I get a hang of streams and generic types in Java. 😅